### PR TITLE
Add prettier to eslint config

### PR DESCRIPTION
**Pull request by darylthayilporch**
Also ran prettier fix on this code base

### CODE REVIEW

**Comment by darylthayilporch** [comment]
https://gitlab.porch.com/frontend/widget-react-api/merge_requests/1093 example of what it looks like in wrapi with prettier (without using eslint config) auto fixed all my files

File: package.json [hunk 1]
```diff
@@ -1,26 +1,30 @@
...
-    "eslint-plugin-mocha": "^4.6.0",
-    "eslint-plugin-react": "^7.5.1"
-  }
+    "name": "@porch/eslint-config-porch",
+    "version": "9.0.0",
+    "description": "Porch's eslint configuration.",
+    "main": "index.js",
+    "scripts": {
+        "lint": "eslint .",
+        "pretty-eslint":  "eslint --fix . && git add . && git status"
```
Comment (jasonnutter) [inline package.json]: Hmm not sure about running `git` commands here.

File: README.md [hunk 2]
```diff
@@ -8,21 +8,83 @@ Porch's [eslint](https://eslint.org/) configuration for JavaScript libraries and
...
+```
+
+If you don't like the defaults, you can rebind editor.action.formatDocument and editor.action.formatSelection in the keyboard shortcuts menu of vscode.
+
+Format On Save
+
+Respects editor.formatOnSave setting.
+
+You can turn off format-on-save on a per-language basis by scoping the setting:
+```
```
Comment (jasonnutter) [inline README.md]:
```
```javascript
```

File: README.md [hunk 3]
```diff
@@ -8,21 +8,83 @@ Porch's [eslint](https://eslint.org/) configuration for JavaScript libraries and
...
+1. Select the text you want to Prettify
+2. CMD + Shift + P -> Format Selection
+Custom keybindings
+```
+
+If you don't like the defaults, you can rebind editor.action.formatDocument and editor.action.formatSelection in the keyboard shortcuts menu of vscode.
+
+Format On Save
+
+Respects editor.formatOnSave setting.
```
Comment (jasonnutter) [inline README.md]: Wrap edit.formatOnSave in back ticks.

File: README.md [hunk 4]
```diff
@@ -8,21 +8,83 @@ Porch's [eslint](https://eslint.org/) configuration for JavaScript libraries and
...
+`eslint . --fix --ext .js --ext .jsx`
+
+### 2. Vscode
+
+Install through VS Code extensions. Search for Prettier - JavaScript formatter.
+Set your vscode setting `prettier.eslintIntegration`to true (defaults to false), this will allow prettier to look at your eslint settings as a base instead of prettier defaults
+
+Using Command Palette (CMD/CTRL + Shift + P)
+
+```
```
Comment (jasonnutter) [inline README.md]: Probably doesn't need to be in a code block.

File: README.md [hunk 5]
```diff
@@ -8,21 +8,83 @@ Porch's [eslint](https://eslint.org/) configuration for JavaScript libraries and
...
+
+Prettier will auto format your code, and you have options how you do this.
+
+### 1. Eslint Fix
+because prettier is implemented with our eslint, you can run this command and all of your files will beformatted according to the eslint / prettier rules
+`eslint . --fix --ext .js --ext .jsx`
+
+### 2. Vscode
+
+Install through VS Code extensions. Search for Prettier - JavaScript formatter.
```
Comment (jasonnutter) [inline README.md]: You should also link to the web page for teh extension.

File: README.md [hunk 6]
```diff
@@ -8,21 +8,83 @@ Porch's [eslint](https://eslint.org/) configuration for JavaScript libraries and
...
+Set your vscode setting `prettier.eslintIntegration`to true (defaults to false), this will allow prettier to look at your eslint settings as a base instead of prettier defaults
```
Comment (jasonnutter) [inline README.md]: VS Code
Prettier
also needs a period at the end.

File: README.md [hunk 7]
```diff
@@ -14,15 +14,76 @@ Porch's [eslint](https://eslint.org/) configuration for JavaScript libraries and
...
+### Precommit
+
+Install the package:
+`yarn add pre-commit --dev`
+
+and add this config to your package.json:
+```
+{
+  "scripts": {
+    "pretty-eslint-precommit":  "eslint --fix . && git add . && git status"
```
Comment (jasonnutter) [inline README.md]: Ah I see.

This should probably be: `npm run eslint -- --fix && git add . && git status`

File: rules/prettier.json [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+  "rules": {
+    "prettier/prettier": [
+      "error",
+      {
+        "singleQuote": true,
+        "tabWidth": 4,
+        "printWidth": 80
+      }
+    ],
+    "react/jsx-sort-props": 0,
```
Comment (jasonnutter) [inline rules/prettier.json]: I think we do want this?

File: package.json [hunk 9]
```diff
@@ -1,26 +1,30 @@
...
-  "homepage": "https://github.com/porchdotcom/eslint-config-porch#readme",
-  "dependencies": {
-    "babel-eslint": "^8.0.2",
-    "eslint": "^4.11.0",
-    "eslint-plugin-import": "^2.8.0",
-    "eslint-plugin-mocha": "^4.6.0",
-    "eslint-plugin-react": "^7.5.1"
-  }
+    "name": "@porch/eslint-config-porch",
+    "version": "9.0.0",
```
Comment (jasonnutter) [inline package.json:3]: You'll need to bump the version, as this package doesn't auto publish. Should probably be `9.1.0`

File: README.md [hunk 10]
```diff
@@ -14,15 +14,76 @@ Porch's [eslint](https://eslint.org/) configuration for JavaScript libraries and
...
+## Prettier
+
+Eslint config now comes bundled with prettier, using `eslint-config-prettier` and `eslint-preset-prettier`.
+
+Prettier overrides some eslint rules and controls styling of your code.
+
+Prettier can auto format your code, and you have options how you do this.
+
+### 1. Eslint Fix
+because prettier is implemented with eslint, you can run this command in your codebase and all of your files will beformatted according to the eslint / prettier rules.
```
Comment (jasonnutter) [inline README.md]: Because

**Review by jasonnutter — CHANGES_REQUESTED** [review]

File: README.md (hunk 7 above)
Comment (darylthayilporch) [reply README.md]: what does the extra -- do ?

File: package.json [hunk 11]
```diff
@@ -1,26 +1,30 @@
 {
-  "name": "@porch/eslint-config-porch",
-  "version": "9.0.0",
```
[the remark this answers is not in the document]
Comment (jasonnutter) [reply package.json:3]: Hmm maybe.

File: package.json (hunk 1 above)
Comment (darylthayilporch) [reply package.json]: I think we want the fix to happen and be added to the commit don't we?

File: README.md (hunk 10 above)
Comment (darylthayilporch) [reply README.md]: fixing

File: package.json (hunk 9 above)
Comment (darylthayilporch) [reply package.json:3]: using npm version ?

File: rules/prettier.json (hunk 8 above)
Comment (darylthayilporch) [reply rules/prettier.json]: whoops probably should not be any rule overrides in here

**Comment by darylthayilporch** [comment]
@jasonn-porch  made updates and asked a few questions

File: package.json (hunk 11 above)
[the remark this answers is not in the document]
Comment (pwmckenna) [reply package.json:3]: I'd advocate for doing this as a minor if we actually want it to be used, given that there's such a low bar to fix everything. We can just run --fix on some of the major repos immediately after publish to eliminate most of the friction

**Comment by pwmckenna** [comment]
I'm really into this. Anything getting our setup closer to the canonical node/react setup the better.

File: package.json (hunk 11 above)
[the remark this answers is not in the document]
Comment (darylthayilporch) [reply package.json:3]: fixing!

**Comment by darylthayilporch** [comment]
updated to minor version, removing WIP

**Comment by darylthayilporch** [comment]
🏋️  updated

**Review by jasonnutter — APPROVED** [review]

**Comment by jasonnutter** [comment]
LGTM

**Pull request merged**